backend/api/student/notifications.py

```
@@ -0,0 +1,310 @@
+    unique_invitations = list({inv.id: inv for inv in all_invitations}.values())
+    
+    return {
+        "invitations": [CohortInvitationResponse.from_orm(inv) for inv in unique_invitations]
+    }
+
+@router.post("/invitations/{invitation_id}/respond")
+async def respond_to_invitation(
+    invitation_id: int,
+    response: InvitationResponse,
+    request,
+    current_user: User = Depends(require_student),
```

Ruff (0.13.1)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)